lib/game.rb

```diff
@@ -0,0 +1,81 @@
+class Game
+    include Interface
+
+    def initialize(player1, player2, board)
+        @player1 = player1
+        @player2 = player2
+        @board = board
+        @status = "initial"
+    end
+
+    def initial
+        [@player1, @player2].each {|player| player.initial}
+        @board.initial
+    end
+
+    def play
```

Comment:
This method works! Great! Though I highly and strongly recommend refactoring it

- It is doing too many things
- It still includes some UI related logic (`puts` and `print`)

What we want perhaps is for play is to simply be:

```ruby
def play
  until game_is_over?
    # 1. ensure player1 makes a move unless game_is_over?
   #  2. ensure player2 makes a move unless game_is_over? 
  end
end
```
Then you will need some `private` methods to help `play`:

```ruby
private

def game_is_over?
  has_won?(player1) || has_won?(player2) || @board.is_full?
end

def has_won?(player)
  # logic that checks whether the moves of player1 satisfy a winning combination
end
```

Reply:
done.